frontend/src/app/create-exercise/create-exercise.component.ts

```diff
@@ -20,7 +20,9 @@ export class CreateExerciseComponent {
   constructor(private router: Router, private route: ActivatedRoute) { }
 
   createExercise(exercise: Exercise): void {
-    exercises.push(exercise);
-    this.router.navigate(['/exercises/' + this.exercise.id]);
+    if(exercise.name !== ""){
```

Comment:
**suggestion:** wrap this logic into a method called `isValid` that you can reuse on the template too so that the logic is contained in a single place